rag/svr/task_executor_refactor/dataset_skill_generator.py

```
@@ -0,0 +1,536 @@
+            title, summary_text, vec = res
+            merged_doc_ids: list[str] = []
+            seen: set[str] = set()
+            merged_doc_texts: dict[str, str] = {}
+            for c in children:
+                for did in c.doc_ids:
+                    if did and did not in seen:
+                        seen.add(did)
+                        merged_doc_ids.append(did)
+                merged_doc_texts.update(c.doc_texts)
+            return SkillNode(
+                level=level + 1,
```

Ruff (0.15.20)

[warning] Function definition does not bind loop variable `level`

(B023)